### DISCUSSION TITLE
The method "isInternational" in class "App\Entity\Conference" requires 0 arguments, but should accept only 1..

### DISCUSSION THREAD

**Discussion by VitaliyMarkelov** [symfony/symfony#54885] [Symfony Book: The Fast Track]
Having followed the instructions up to https://symfony.com/doc/6.4/the-fast-track/en/9-backend.html
_You can now add/modify/delete conferences directly from the admin backend. Play with it and add at least one conference._
I have encountered a problem. In the attempt to save first conference I got an error, stack trace is below. The method concerned is this:
```
    public function isInternational(): ?bool
    {
        return $this->isInternational;
    }
```
 any advice is appreciated.

Symfony\Component\PropertyAccess\Exception\NoSuchPropertyException:
The method "isInternational" in class "App\Entity\Conference" requires 0 arguments, but should accept only 1..

  at C:\PROJECTS\guestbook\vendor\symfony\property-access\PropertyAccessor.php:539
  at Symfony\Component\PropertyAccess\PropertyAccessor->writeProperty(array(object(Conference)), 'isInternational', false)
     (C:\PROJECTS\guestbook\vendor\symfony\property-access\PropertyAccessor.php:120)
  at Symfony\Component\PropertyAccess\PropertyAccessor->setValue(object(Conference), object(PropertyPath), false)
     (C:\PROJECTS\guestbook\vendor\symfony\form\Extension\Core\DataAccessor\PropertyPathAccessor.php:75)
  at Symfony\Component\Form\Extension\Core\DataAccessor\PropertyPathAccessor->setValue(object(Conference), false, object(Form))
     (C:\PROJECTS\guestbook\vendor\symfony\form\Extension\Core\DataAccessor\ChainAccessor.php:48)
  at Symfony\Component\Form\Extension\Core\DataAccessor\ChainAccessor->setValue(object(Conference), false, object(Form))
     (C:\PROJECTS\guestbook\vendor\symfony\form\Extension\Core\DataMapper\DataMapper.php:73)
  at Symfony\Component\Form\Extension\Core\DataMapper\DataMapper->mapFormsToData(object(RecursiveIteratorIterator), object(Conference))
     (C:\PROJECTS\guestbook\vendor\symfony\form\Form.php:551)
  at Symfony\Component\Form\Form->submit(array(), true)
     (C:\PROJECTS\guestbook\vendor\symfony\form\Extension\HttpFoundation\HttpFoundationRequestHandler.php:110)
  at Symfony\Component\Form\Extension\HttpFoundation\HttpFoundationRequestHandler->handleRequest(object(Form), object(Request))
     (C:\PROJECTS\guestbook\vendor\symfony\form\Form.php:420)
  at Symfony\Component\Form\Form->handleRequest(object(Request))
     (C:\PROJECTS\guestbook\vendor\easycorp\easyadmin-bundle\src\Controller\AbstractCrudController.php:305)
  at EasyCorp\Bundle\EasyAdminBundle\Controller\AbstractCrudController->new(object(AdminContext))
     (C:\PROJECTS\guestbook\vendor\symfony\http-kernel\HttpKernel.php:181)
  at Symfony\Component\HttpKernel\HttpKernel->handleRaw(object(Request), 1)
     (C:\PROJECTS\guestbook\vendor\symfony\http-kernel\HttpKernel.php:76)
  at Symfony\Component\HttpKernel\HttpKernel->handle(object(Request), 1, true)
     (C:\PROJECTS\guestbook\vendor\symfony\http-kernel\Kernel.php:197)
  at Symfony\Component\HttpKernel\Kernel->handle(object(Request))
     (C:\PROJECTS\guestbook\vendor\symfony\runtime\Runner\Symfony\HttpKernelRunner.php:35)
  at Symfony\Component\Runtime\Runner\Symfony\HttpKernelRunner->run()
     (C:\PROJECTS\guestbook\vendor\autoload_runtime.php:29)
  at require_once('C:\\PROJECTS\\guestbook\\vendor\\autoload_runtime.php')
     (C:\PROJECTS\guestbook\public\index.php:5)

**Comment by MatTheCat** [comment]
Does your `Conference` entity expose a setter for `isInternational`? Can you paste its code here?

**Reply by VitaliyMarkelov** [reply]
It does, at least all methods related to $isInternational are the same as in [repo:](https://github.com/the-fast-track/book-6.2-1/blob/main/src/Entity/Conference.php)
```
<?php

namespace App\Entity;

use App\Repository\ConferenceRepository;
use Doctrine\Common\Collections\ArrayCollection;
use Doctrine\Common\Collections\Collection;
use Doctrine\ORM\Mapping as ORM;

#[ORM\Entity(repositoryClass: ConferenceRepository::class)]
class Conference
{
    #[ORM\Id]
    #[ORM\GeneratedValue]
    #[ORM\Column]
    private ?int $id = null;

    #[ORM\Column(length: 255)]
    private ?string $city = null;

    #[ORM\Column(length: 4)]
    private ?string $year = null;

    #[ORM\Column]
    private ?bool $isInternational = null;

    /**
     * @var Collection<int, Comment>
     */
    #[ORM\OneToMany(targetEntity: Comment::class, mappedBy: 'conference', orphanRemoval: true)]
    private Collection $comments;

    public function __construct()
    {
        $this->comments = new ArrayCollection();
    }


	public function __toString(): string
    {
        return $this->city.' '.$this->year;
    }

    public function getId(): ?int
    {
        return $this->id;
    }

    public function getCity(): ?string
    {
        return $this->city;
    }

    public function setCity(string $city): static
    {
        $this->city = $city;

        return $this;
    }

    public function getYear(): ?string
    {
        return $this->year;
    }

    public function setYear(string $year): static
    {
        $this->year = $year;

        return $this;
    }

    public function isInternational(): ?bool
    {
        return $this->isInternational;
    }

    public function setInternational(bool $isInternational): static
    {
        $this->isInternational = $isInternational;

        return $this;
    }

    /**
     * @return Collection<int, Comment>
     */
    public function getComments(): Collection
    {
        return $this->comments;
    }

    public function addComment(Comment $comment): static
    {
        if (!$this->comments->contains($comment)) {
            $this->comments->add($comment);
            $comment->setConference($this);
        }

        return $this;
    }

    public function removeComment(Comment $comment): static
    {
        if ($this->comments->removeElement($comment)) {
            // set the owning side to null (unless already changed)
            if ($comment->getConference() === $this) {
                $comment->setConference(null);
            }
        }

        return $this;
    }
}
```

**Reply by MatTheCat** [reply]
I guess your setter should be `setIsInternational` (and not `setInternational`) as your property is called `isInternational`?

**Reply by michaljusiega** [reply]
And be careful, when property has equal-name accessor-method (setter/getter). In your case you have `isInternational` method like your property `$isInternational`. See https://github.com/symfony/symfony/issues/54488 for more.

**Reply by VitaliyMarkelov** [reply]
@MatTheCat I guess the same, but tutorial's class doesn't have it either. Anyway, I wiped this and started from scratch. Thank you for your help.

**Reply by VitaliyMarkelov** [reply]
@michaljusiega I would never use such a risky name for a property myself, but this isn't my idea, it is official SF [book's](https://symfony.com/doc/6.4/the-fast-track/en/index.html) one ).

**Comment by ghost** [comment]
I have the same error. Changing `setInternational` to `setIsInternational` solved it.

**Reply by Bassinga1** [reply]
Me too I have: public function setActive(bool $isActive): static // Solution: public function setIsActive(bool $isActive): static

**Comment by toby-griffiths** [comment]
So this looks like a bug with the Maker that generates these files, as I have this problem, and I generated these using the maker.

**Reply by MatTheCat** [reply]
Yep guess it comes from https://github.com/symfony/maker-bundle/pull/1493

**Reply by toby-griffiths** [reply]
I see a comment on https://github.com/symfony/maker-bundle/pull/1493, after it was merged, about it breaking the property accessor, but I cannot find an issue raised to address the property accessor.

Is anyone aware of this being dealt with, or should I raise a bug for that?

**Reply by linuxprocess** [reply]
I just run into this problem, could you fill a rug report please ?

**Reply by Vincere1st** [reply]
I've encountered the same issue this day, and i have not find issue open. Anyone has open an issue?

**Reply by irzadi** [reply]
Sure hope they at least added some notice / warning to the book.
Someone learning the framework for the first time would simply think he is doing something wrong instead of blaming the library.
Wasted a couple hours redoing the app twice this morning due to this bug.

**Reply by stof** [reply]
the bug has been fixed in MakerBundle, but is not released yet

**Comment by Pyrrah** [comment]
Thanks for the feedback, I solved my problem. 
As I write, it's not fixed.

The subject is being corrected ?
